### DISCUSSION TITLE
Fonctionnalités à implémenter

### DISCUSSION THREAD

**Discussion by Fred-si** [devcherchecollegue-org/devguy-python#4] [Ideas]
Salut tout le monde, je vous propose qu’on discute ici des premières fonctionnalités à implémenter. Pour l’instant le bot actuel est utilisé pour

> 1. Auto-attribution de rôles par les membres avec le système de réactions visible sur #welcome .
> 1. Message d’accueil personnalisable pour les nouveaux arrivants sur le serveur.
> 1. Attribution d’Xp, avec système de niveaux personnalisables.
> 1. Commandes diverses et variées, que je n’exploite pas du tout, comme bcp d’autres choses qu’est censé savoir faire Devbot.
(dixit Octavius)

qu’en pensez-vous ?

**Comment by valerie3111** [comment]
Ça m'a l'air pas mal pour commencer

**Comment by Maestro31** [comment]
Voilà mes remarques :

1 - Le système est plus complexe qu'un simple message. MEE6 utilise l'api de Discord pour faire la liste des channels avant envoi d'un message. Ce message peut être créé, édité et supprimé depuis un backoffice dédié pour lequel on est connecté via oauth à notre compte Discord. Donc, si on veut le même système, il faut disposer d'un serveur web (stack libre puisqu'API disponible), un autre repo pour le code de ce serveur, serveur de test,... (ça c'est si on veut la version complète, iso à MEE6)

2 - Cette partie devrait être rapide à faire, à moins qu'on veuilles permettre la personnalisation de ce message (serveur web, API,...)

3 - Même problématique que 1, serveur à faire avec backoffice,...

4 - Il faudrait expliciter cette partie pour qu'on comprenne mieux de quoi il s'agit.

En ce qui concerne la faisabilité générale, c'est tout à fait réalisable mais avec pas mal de challenge.
